### ISSUE TITLE
No inputs are created

### ISSUE THREAD

**Issue by Kinzal16** [TG908/homebridge-vsx#17]
Hi @TG908, I have installed "homebridge-vsx-2". My SC-2022 is found and also displayed in the Apple Home app, but I can only switch it on and off. The channels inputs are missing

**Comment by TG908** [comment]
Changing Input channels is not yet implemented. The only thing you could do is try to reverse engineer the iControl AV App and implement a way to switch channels. But don't even know if Home Kit or homebridge currently supports a "channel" characteristic.

**Comment by Kinzal16** [comment]
Yes, Homebridge and Homekit support channel swapping.

![IMG_5787](https://user-images.githubusercontent.com/22612289/86834682-624e8b80-c09b-11ea-8af9-8f44bfdd77ef.PNG)

**Comment by TG908** [comment]
Good to know. I did some more research and found out that homebridge also support the right characteristics. The characteristic is meant to be used with TVs but I think this would work for av receivers as well. I unfortunately cannot help you with implementing this since I no longer have access to a pioneer av receiver. But feel free to ask me anything if you need help.

**Comment by Kinzal16** [comment]
Yes, it would work with the AVR too. It works e.g. also with the Logitech Harmony.

I'm not a programmer, I can only test what they do. Maybe you can implement an input, then I can test it. If it works, additional inputs could be implemented. You could also set the inputs as switches.

I would be very happy if you would.

**Comment by TG908** [comment]
To implement that I need to reverse engineer the app. This includes capturing network traffic between the devices.

**Comment by Kinzal16** [comment]
That would be e.g. the command for "Stereo" 0009SR

**Comment by fearlesschicken** [comment]
Aren't we able to change (and query) the input from telnet connection?
https://raymondjulin.com/blog/remote-control-your-pioneer-vsx-receiver-over-telnet
